src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/Switch-AzureSqlDatabaseInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,145 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Services;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.InstanceFailoverGroup.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Switch, "AzureRmSqlDatabaseInstanceFailoverGroup",
+        ConfirmImpact = ConfirmImpact.Medium, SupportsShouldProcess = true)]
+    public class SwitchAzureSqlInstanceFailoverGroup : AzureSqlInstanceFailoverGroupCmdletBase
+    {
+        /// <summary>
+        /// Parameter set name for switch with an Input Object.
+        /// </summary>
+        protected const string SwitchIFGByInputObjectParameterSet =
+            "Switch a Instance Failover Group from AzureSqlInstanceFailoverGroupModel instance definition";
+        
+        /// <summary>
+        /// Parameter set name for switch with a resource ID.
+        /// </summary>
+        private const string SwitchIFGByResourceIdSet = "Switch a Instance Failover Group from Resource Id";
+
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@jugeorge same comment about `ValueFromPipelineByPropertyName` and moving the parameters into a separate parameter set